structured_type_read_test.go

```diff
@@ -1621,3 +1622,156 @@ func forAllStructureTypeFormats(dbt *DBTest, f func(t *testing.T, format string)
 		})
 	}
 }
+
+func TestSelectingNullObjectsInArrowBatches(t *testing.T) {
+	testcases := []string{
+		"select null::object(v VARCHAR)",
+		"select null::object",
+	}
+	skipStructuredTypesTestsOnGHActions(t)
+	runDBTest(t, func(dbt *DBTest) {
+		for _, tc := range testcases {
+			t.Run(tc, func(t *testing.T) {
+				pool := memory.NewCheckedAllocator(memory.DefaultAllocator)
+				defer pool.AssertSize(t, 0)
+				ctx := WithArrowBatches(WithArrowAllocator(context.Background(), pool))
+
+				var err error
+				var rows driver.Rows
+				err = dbt.conn.Raw(func(sc any) error {
+					queryer, implementsQueryContext := sc.(driver.QueryerContext)
+					assertTrueF(t, implementsQueryContext, "snowflake conn does not implement QueryerContext but needs to")
+					rows, err = queryer.QueryContext(ctx, tc, nil)
+					return err
+				})
+				assertNilF(t, err, fmt.Sprintf("test failed to run the following query: %s", tc))
+				defer rows.Close()
+
+				sfRows, isSfRows := rows.(SnowflakeRows)
+				assertTrueF(t, isSfRows, "rows are not snowflakeRows")
+
+				batches, err := sfRows.GetArrowBatches()
+				assertNilF(t, err)
+				assertNotEqualF(t, len(batches), 0)
+				batch, err := batches[0].Fetch()
+				assertNilF(t, err)
+				assertNotEqualF(t, len(*batch), 0)
+				for _, record := range *batch {
+					// check number of cols/rows so we dont get index out of range
+					assertEqualF(t, record.NumRows(), int64(1), "wrong number of rows")
+					assertEqualF(t, record.NumCols(), int64(1), "wrong number of cols")
+
+					colIndex := 0
+					rowIndex := 0
+
+					srtCol, isStrCol := record.Column(colIndex).(*array.String)
+					assertTrueF(t, isStrCol, "column is not type string")
+
+					assertEqualE(t, srtCol.Value(rowIndex), "")
+					isNull := srtCol.IsNull(rowIndex)
+					assertTrueF(t, isNull)
+					record.Release()
+				}
+
+			})
+		}
+	})
+}
+
+func TestSelectingSemistructuredObjectInArrowBatches(t *testing.T) {
```

Comment:
If a test contains arrays as well as objects, maybe we can call it `semistructuredTypes`?

Reply:
Updated!